spec/system/grant_submissions/grant_submission_responses_spec.rb

```diff
@@ -8,137 +8,342 @@ def successful_application_message
+                click_button 'Submit'
+              end
 
-          click_button 'Submit'
-          expect(page).not_to have_content successful_application_message
+              scenario 'do not receive success message on field with wrong type of response' do
+                expect(page).not_to have_content successful_application_message
+                expect(page).to have_link 'text_file.pdf'
+              end
+
+              scenario 'submits file successfully when file when validation error is corrected' do
+                find_field('Number Question', with: 'Number').set(10)
+                click_button 'Submit'
+                expect(page).to have_content successful_application_message
+                expect(GrantSubmission::Response.find_by(question: file_upload_question).document.attached?).to be true
+              end
+            end
+          end
         end
       end
 
-      context 'combined questions' do
-        context 'file_upload' do
-          context 'with another invalid response' do
+      context 'states' do
+        let(:file_upload_question) {@grant.questions.find_by(response_type: 'file_upload')}
+
+        context 'required questions' do
+          context 'on create' do
+            context 'saving as draft' do
+              context 'requires response mandatory short_text question' do
+                scenario 'no answer for required short text' do
+                  short_text_question = @grant.questions.where(response_type: 'short_text').first
+                  short_text_question.update_attribute(:is_mandatory, true)
+
+                  find_field('Number Question', with:'').set(Faker::Number.number(digits: 10))
+                  find_field('Long Text Question', with:'').set(Faker::Lorem.paragraph_by_chars(number: 1000))
+                  click_button 'Save as Draft'
+                  expect(page).to have_content 'You started your submission.'
+                  expect(page).to have_current_path profile_submissions_path
+                  expect(page).to have_content GrantSubmission::Submission.last.title
+                  expect(page).to have_content 'Edit'
+                end
+              end
+            end
+
+            context 'submitting' do
+              context 'requires response mandatory short_text question' do
+                scenario 'no answer for required short text' do
+                  short_text_question = @grant.questions.where(response_type: 'short_text').first
+                  short_text_question.update_attribute(:is_mandatory, true)
+
+                  find_field('Number Question', with:'').set(Faker::Number.number(digits: 10))
+                  find_field('Long Text Question', with:'').set(Faker::Lorem.paragraph_by_chars(number: 1000))
+                  click_button 'Submit'
+                  expect(page).to have_content 'Please review the following error'
+                  expect(page).to have_content 'is required'
```

Comment:
Are you getting the right 'is required' error? e.g. Is it supposed to include the short_text_question.text?

Reply:
This should test for a specific error. See other examples above.